src/Symfony/Component/Form/FormRenderer.php

```diff
@@ -132,6 +132,9 @@ public function searchAndRenderBlock(FormView $view, $blockNameSuffix, array $va
         $renderOnlyOnce = 'row' === $blockNameSuffix || 'widget' === $blockNameSuffix;
 
         if ($renderOnlyOnce && $view->isRendered()) {
+            // This is not allowed, because it would result in rendering same IDs multiple times, which is not valid.
+            @trigger_error(sprintf('Calling the "%s()" method for fields which were already rendered is deprecated since Symfony 4.1 and will throw an exception in 5.0.', __METHOD__), E_USER_DEPRECATED);
```

Comment:
should be 4.2 here.

Reply:
Can we add the field name here as well to help figuring out where that comes from in an app?

Reply:
people don't call `searchAndRenderBlock` directly, so this deprecation is confusing as they probably don't know about it.

Reply:
@stof What do you suggest? I agree people don't call this directly, but formRenderer is responsible for this confusing behavior so deprecation needs to be triggered from within.

Reply:
well, `$blockNameSuffix` allows you to know whether `form_row` or `form_widget` was called originally, so you can use it to report it.

and I agree about the field name